gemrb/core/GameScript/GameScript.cpp

```diff
@@ -1659,6 +1659,49 @@ static void SetupSavedTriggers()
 	}
 }
 
+static std::string ReadNamedRowColumnIndex(const AutoTable& table, const TableMgr::key_t& rowName, int columnIndex)
+{
+	TableMgr::index_t rowIndex = table->GetRowIndex(rowName);
+	return rowIndex != TableMgr::npos ? table->QueryField(rowIndex, columnIndex) : table->QueryDefault();
+}
+
+static int ReadNamedRowColumnIndexSigned(const AutoTable& table, const TableMgr::key_t& rowName, int columnIndex)
+{
+	return strtosigned<int>(ReadNamedRowColumnIndex(table, rowName, columnIndex).c_str());
+}
+
+static void InitializeObjectIDS(const AutoTable& objNameTable)
+{
+	TableMgr::index_t idsRow = objNameTable->GetRowIndex("OBJECT_IDS_COUNT");
+
+	ObjectIDSCount = objNameTable->QueryFieldSigned<int>(idsRow, 0);
+	if (ObjectIDSCount < 0 || ObjectIDSCount > MAX_OBJECT_FIELDS) {
+		error("GameScript", "The IDS Count shouldn't be more than 10!");
+	}
+
+	ObjectIDSTableNames.resize(ObjectIDSCount);
+	for (int i = 0; i < ObjectIDSCount; i++) {
+		const std::string& idsName = objNameTable->QueryField(idsRow, i + 1);
+		const IDSLink* poi = FindIdentifier(idsName.c_str());
+		if (poi == nullptr) {
+			idtargets[i] = nullptr;
+		} else {
+			idtargets[i] = poi->Function;
+		}
+		ObjectIDSTableNames[i] = ResRef(idsName);
+	}
+
+	MaxObjectNesting = ReadNamedRowColumnIndexSigned(objNameTable, "MAX_OBJECT_NESTING", 0);
```

Comment:
Why not just pass the column by name ("DATA")? That way it looks like you could avoid the extra functions.

Reply:
Good call - I have a bit of a tendency towards micro-optimization. I've replaced these calls with standard named lookups.

Reply:
We actually prefer string-based table lookups, since then it's much easier to change the structure of 2das (or data type) if needed. It will be relevant again for some EE tables like clasiskl.2da (duplicates part of our classes.2da/clskills.2da), since we'll want to unify the approach / layouts.